--- J:\Projects\RiscV\riscv-tools\riscv-gnu-toolchain\riscv-gcc\gcc\config.gcc
+++ J:\Snippets\Fixes\riscv32\gcc\config.gcc
@@ -2032,10 +2036,14 @@
 	gcc_cv_initfini_array=yes
 	;;
 riscv*-*-elf* | riscv*-*-rtems*)
+	case "${target}" in
+	riscv32-*) riscv_xlen=32 ;;
+	riscv64-*) riscv_xlen=64 ;;
+	esac
 	tm_file="elfos.h newlib-stdint.h ${tm_file} riscv/elf.h"
 	case "x${enable_multilib}" in
 	xno) ;;
-	xyes) tmake_file="${tmake_file} riscv/t-elf-multilib" ;;
+	xyes) tmake_file="${tmake_file} riscv/t-elf-multilib${riscv_xlen}" ;;
 	*) echo "Unknown value for enable_multilib"; exit 1
 	esac
 	tmake_file="${tmake_file} riscv/t-riscv"

